example/src/main/java/com/basistheory/android/example/view/reveal/RevealDataFragment.kt

```diff
@@ -0,0 +1,105 @@
+package com.basistheory.android.example.view.reveal
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.fragment.app.Fragment
+import androidx.fragment.app.viewModels
+import com.basistheory.Token
+import com.basistheory.android.example.databinding.FragmentRevealBinding
+import com.basistheory.android.example.util.tokenExpirationTimestamp
+import com.basistheory.android.example.viewmodel.CardFragmentViewModel
+import com.basistheory.android.model.ElementValueReference
+import com.basistheory.android.service.ProxyRequest
+import com.basistheory.android.service.getElementValueReference
+
+class RevealDataFragment : Fragment() {
+    private val binding: FragmentRevealBinding by lazy {
+        FragmentRevealBinding.inflate(layoutInflater)
+    }
+    private val viewModel: CardFragmentViewModel by viewModels()
+    private var tokenId: String? = null
+
+    override fun onCreateView(
+        inflater: LayoutInflater,
+        container: ViewGroup?,
+        savedInstanceState: Bundle?
+    ): View {
+        super.onCreateView(inflater, container, savedInstanceState)
+        binding.lifecycleOwner = this
+        binding.viewModel = viewModel
+
+        binding.cvc.cardNumberElement = binding.cardNumber
+
+        binding.tokenizeButton.setOnClickListener { tokenize() }
+        binding.revealButton.setOnClickListener { reveal() }
+
+        setValidationListeners()
+
+        return binding.root
+    }
+
+    private fun tokenize() =
+        viewModel.tokenize(
+            object {
+                val type = "card"
+                val data = object {
+                    val number = binding.cardNumber
+                    val expiration_month = binding.expirationDate.month()
+                    val expiration_year = binding.expirationDate.year()
+                    val cvc = binding.cvc
+                }
+                val expires_at = tokenExpirationTimestamp()
+            }).observe(viewLifecycleOwner) {
+            tokenId = (it as Map<*, *>)["id"] as String
```

Comment:
Maybe it could even use a generic type argument so we can attempt to cast for them 🤷 :
```kotlin
it.getValue<String>("id")
it.getValue<UUID>("created_by")
```